dandiapi/zarr/tests/test_zarr_upload.py

```diff
@@ -4,15 +4,22 @@
 from storages.backends.s3 import S3Storage
 from zarr_checksum.checksum import EMPTY_CHECKSUM
 
+from dandiapi.api.models.dandiset import Dandiset
 from dandiapi.api.services.permissions.dandiset import add_dandiset_owner
 from dandiapi.api.tests.fuzzy import HTTP_URL_RE
 from dandiapi.zarr.models import ZarrArchive, ZarrArchiveStatus
 
 
 @pytest.mark.django_db
+@pytest.mark.parametrize('embargoed', [False, True])
 def test_zarr_rest_upload_start(
-    authenticated_api_client, user, zarr_archive: ZarrArchive, storage, monkeypatch
+    authenticated_api_client, user, zarr_archive_factory, storage, monkeypatch, embargoed
 ):
+    zarr_archive = zarr_archive_factory(
```

Comment:
If you're going to start using a factory, you can move the mutations to `zarr_archive` below into it too.